amazon/ion/simpleion.py

```diff
@@ -470,10 +477,11 @@ def add(obj):
         event = reader.send(NEXT_EVENT)
 
 
-def dump_extension(obj, fp, binary=True, sequence_as_stream=False, tuple_as_sexp=False, omit_version_marker=False):
+def dump_extension(obj, fp, binary=True, sequence_as_stream=False, tuple_as_sexp=False, omit_version_marker=False,
+                   trailing_commas=False):
     """C-extension implementation. Users should prefer to call ``dump``."""
 
-    res = ionc.ionc_write(obj, binary, sequence_as_stream, tuple_as_sexp)
+    res = ionc.ionc_write(obj, binary, sequence_as_stream, tuple_as_sexp, trailing_commas)
```

Comment:
This call is what is causing the segfault in the unit tests. `ionc_write` isn't expecting to get the trailing_commas argument, and is not handling it well.

If you want to put in a `# TODO:`, and drop the `trailing_commas` from the `ionc.ionc_write` call, I can dig into why it isn't handling that a bit more gracefully, and that should clear up the test errors (the ones that aren't already explained).

Reply:
Wait, I must be confused. I'm not passing the arg to the C code, as far as I can see?
(Note lack of `trailing_commas=trailing_commas` in the first branch.)

```
    if c_ext and __IS_C_EXTENSION_SUPPORTED and (imports is None and indent is None):
        return dump_extension(obj, fp, binary=binary, sequence_as_stream=sequence_as_stream,
                              tuple_as_sexp=tuple_as_sexp, omit_version_marker=omit_version_marker)
    else:
        return dump_python(obj, fp, imports=imports, binary=binary, sequence_as_stream=sequence_as_stream,
                           indent=indent,
                           tuple_as_sexp=tuple_as_sexp, omit_version_marker=omit_version_marker,
                           trailing_commas=trailing_commas)
```

Reply:
Oh, oops, in `dump_extension`. D'oh.

Reply:
Done.

(Removed `trailing_commas` from `dump_extension()` and added a `TODO`).